app/models/host/discovered.rb

```diff
@@ -81,6 +81,8 @@ def populate_fields_from_facts(facts = self.facts_hash, ignored_type)
     unless primary_ip.nil?
       subnet = Subnet.subnet_for(primary_ip)
       if subnet
+        # reload subnet to fetch all references
+        subnet.reload
```

Comment:
this will trigger a query which wasn't triggered before. don't understand how that brings number of queries down...